### PR TITLE
Migrate to Redux Toolkit part 8

### PR DESCRIPTION
**Pull request by RichDom2185**
### Description

Migrated more sagas and action creators to use the new helpers.

Codebase is still a bit messy as it's in the midst of refactoring but it will be cleaned up in subsequent parts.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Code quality improvements

### How to test

Everything should still work.

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

File: src/commons/sagas/__tests__/BackendSaga.ts [hunk 1]
```diff
@@ -1167,11 +1163,13 @@ describe('Test ADD_NEW_USERS_TO_COURSE action', () => {
       .not.put.actionType(fetchAdminPanelCourseRegistrations.type)
       .not.call.fn(showSuccessMessage)
       .provide([[call(putNewUsers, mockTokens, users, provider), errorResp]])
-      .dispatch({ type: ADD_NEW_USERS_TO_COURSE, payload: { users, provider } })
+      .dispatch({ type: addNewUsersToCourse.type, payload: { users, provider } })
       .silentRun();
   });
 });
 
+// TODO: Test addNewStoriesUsersToCourse
```
Comment (chownces) [inline src/commons/sagas/__tests__/BackendSaga.ts:1171]: Untested

File: src/commons/workspace/__tests__/WorkspaceActions.ts [hunk 2]
```diff
@@ -378,12 +334,14 @@ test('setEditorHighlightedLines generates correct action object', () => {
   });
 });
 
+// TODO: Test setEditorHighlightedLinesControl
```
Comment (chownces) [inline src/commons/workspace/__tests__/WorkspaceActions.ts:337]: .

File: src/commons/workspace/__tests__/WorkspaceActions.ts [hunk 3]
```diff
@@ -656,10 +614,12 @@ test('updateChapter generates correct action object', () => {
...
-    type: TOGGLE_USING_SUBST,
+    type: toggleUsingSubst.type,
     payload: {
       workspaceLocation: playgroundWorkspace,
       usingSubst: true
     }
   });
 });
+
+// TODO: Add toggleusingcse
```
Comment (chownces) [inline src/commons/workspace/__tests__/WorkspaceActions.ts:625]: .

File: src/commons/sagas/WorkspaceSaga/index.ts [hunk 4]
```diff
@@ -514,15 +469,12 @@ export default function* WorkspaceSaga(): SagaIterator {
...
         );
       }
-    }
-  );
-
-  yield takeLeading(
-    EVAL_EDITOR_AND_TESTCASES,
-    function* (action: ReturnType<typeof actions.runAllTestcases>) {
+    },
+    // TODO: Should be takeLeading, not takeEvery
```
Comment (chownces) [inline src/commons/sagas/WorkspaceSaga/index.ts:473]: Was this behaviour checked?

**Review by chownces — APPROVED** [review]
Some minor comments. Thanks for the continual work on the refactor!

File: src/commons/sagas/WorkspaceSaga/index.ts (hunk 4 above)
Comment (RichDom2185) [reply src/commons/sagas/WorkspaceSaga/index.ts:473]: Will have to test with an actual assessment with testcases. But should be no different to `takeEvery` except for the fact that `takeLeading` is blocking.

**Pull request merged**